src/lib/opal_parser.c

```diff
@@ -160,26 +180,27 @@ int append_bytes(uint8_t *buf, size_t len, const uint8_t *src, int src_len)
 	int ret = 0;
 
 	/* data fits into opal short atom */
-	if (src_len <= 15) {
-		if (len >= src_len + 1) {
+	if (src_len < 16) {
+		if (len >= src_len + 1)
 			ret += append_short_atom_bytes_header(buf, len, src_len);
-			memcpy(buf + ret, src, src_len);
-			ret += src_len;
-		}
+
 	/* data fits into opal medium atom */
-	}
-	else if (src_len <= 2047) {
-		if (len >= src_len + 2) {
+	} else if (src_len < 2048) {
+		if (len >= src_len + 2)
 			ret += append_medium_atom_bytes_header(buf, len, src_len);
-			memcpy(buf + ret, src, src_len);
-			ret += src_len;
-		}
+
+	/* data fits into opal long atom */
+	} else if (src_len < 16777216) {
+		if (len >= src_len + 4)
+			ret += append_long_atom_bytes_header(buf, len, src_len);
 	}
```

Comment:
please add assert for } else { buffer sizes

Reply:
I was planning on introducing #defines for the max short, medium and long atom data lens.
Also, insert an assert at the beginning of the function rather than having it in an else condition.
assert (src_len < MAX_LONG_ATOM_DATA_LEN)
What do you think ?